.github/workflows/scripts/auto-update-version.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash -e
+
+repo=$1
+variable_name=$2
+file_names=("${@:3}") # remaining args
+
+latest_version=$(gh api -q .tag_name "repos/open-telemetry/$repo/releases/latest" | sed 's/^v//')
+
+echo "Repo: $repo"
+echo "Latest version: $latest_version"
+
+for file_name in "${file_names[@]}"
+do
+  sed -i -e "s/$variable_name: .*/$variable_name: $latest_version/" "$file_name"
+done
+
+if git diff --quiet; then
+  echo "Already at the latest version."
+  exit 0
+fi
+
+message="Update $repo version to $latest_version"
+body="Update $repo version to \`$latest_version\`."
+
+existing_pr_count=$(gh pr list --state all --search "in:title $message" | wc -l)
+if [ "$existing_pr_count" -gt 0 ]; then
+    echo "PR for this version was already created."
+    exit 0
+fi
+
+branch="opentelemetrybot/auto-update-$repo-$latest_version"
```

Comment:
I'd drop opentelemetrybot, because I want this to be executable locally if/when we might feel the need for an immediate update. If you prefer having a branch of the form `foo/bar` then maybe it can be `auto-update/...`. Otherwise a simple branch seems fine, WDYT?

```suggestion
branch="auto-update-$repo-$latest_version"
```

Reply:
> if/when we might feel the need for an immediate update

this workflow supports `workflow_dispatch` (see top), which means you can trigger it manually in GitHub Actions any time you want (this is also a really handy way to test the workflow on your fork, push to `main` on your fork, and then run manually trigger the workflow)

Reply:
Got it, thanks!